exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -115,6 +118,29 @@ const (
 	Stable ReleaseChannel = "stable"
 )
 
+// MasterAuthorizedNetworksConfig contains configuration options for the master authorized networks feature.
+// Enabled master authorized networks will disallow all external traffic to access
+// Kubernetes master through HTTPS except traffic from the given CIDR blocks,
+// Google Compute Engine Public IPs and Google Prod IPs.
+type MasterAuthorizedNetworksConfig struct {
+	// Whether or not master authorized networks is enabled.
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
As **MasterAuthorizedNetworksConfig** is optional on  `GCPManagedControlPlaneSpec` then i would say that if someone specifies a non-nil **MasterAuthorizedNetworksConfig** then this means they want to enable master authorized networks. So i feel that the **Enabled** field is not needed.

Reply:
wdyt @cpanato ?

Reply:
Personally I would prefer that we make `MasterAuthorizedNetworksConfig == nil` to  mean disabled.